yt_dlp/extractor/vrt.py

```diff
@@ -1,82 +1,216 @@
+                                             video_id, note='refreshtoken: Retrieve vrtnutoken',
+                                             errnote='refreshtoken failed')['vrtnutoken']
+
+        headers = self.geo_verification_headers()
+        headers.update({'Content-Type': 'application/json; charset=utf-8'})
+        vrtPlayerToken = self._download_json(
+            '%s/tokens' % self._REST_API_BASE_TOKEN, video_id,
+            'Downloading token', headers=headers, data=json.dumps({
+                'identityToken': vrtnutoken
+            }).encode('utf-8'))['vrtPlayerToken']
+
+        video_info = self._download_json(
+            '%s/media-items/%s' % (self._REST_API_BASE_VIDEO, video_id),
+            video_id, 'Downloading video JSON', query={
+                'vrtPlayerToken': vrtPlayerToken,
+                'client': 'vrtnu-web@PROD',
+            }, expected_status=400)
+        if 'title' not in video_info:
+            code = video_info.get('code')
+            if code == 'AUTHENTICATION_REQUIRED':
+                self.raise_login_required()
+            elif code == 'INVALID_LOCATION':
+                self.raise_geo_restricted(countries=['BE'])
+            raise ExtractorError(video_info.get('message') or code, expected=True)
+
+        formats = []
+        subtitles = {}
+        for target in video_info['targetUrls']:
+            format_url, format_type = url_or_none(target.get('url')), str_or_none(target.get('type'))
+            if not format_url or not format_type:
+                continue
+            format_type = format_type.upper()
+            if format_type in self._HLS_ENTRY_PROTOCOLS_MAP:
+                fmts, subs = self._extract_m3u8_formats_and_subtitles(
+                    format_url, video_id, 'mp4', self._HLS_ENTRY_PROTOCOLS_MAP[format_type],
+                    m3u8_id=format_type, fatal=False)
+                formats.extend(fmts)
+                subtitles = self._merge_subtitles(subtitles, subs)
+            elif format_type == 'HDS':
+                formats.extend(self._extract_f4m_formats(
+                    format_url, video_id, f4m_id=format_type, fatal=False))
+            elif format_type == 'MPEG_DASH':
+                fmts, subs = self._extract_mpd_formats_and_subtitles(
+                    format_url, video_id, mpd_id=format_type, fatal=False)
+                formats.extend(fmts)
+                subtitles = self._merge_subtitles(subtitles, subs)
+            elif format_type == 'HSS':
+                fmts, subs = self._extract_ism_formats_and_subtitles(
+                    format_url, video_id, ism_id='mss', fatal=False)
+                formats.extend(fmts)
+                subtitles = self._merge_subtitles(subtitles, subs)
+            else:
+                formats.append({
+                    'format_id': format_type,
+                    'url': format_url,
+                })
+        self._sort_formats(formats)
```

Comment:
Once this branch gets rebased against master this would no longer be needed ([source](https://github.com/yt-dlp/yt-dlp/commit/9f14daf22b4080ae1531a772ee7574959af4e2fa)):

```suggestion
```